packages/ui-components/src/components/DateTimePicker/DateTimePicker.component.tsx

```diff
@@ -0,0 +1,700 @@
+  }, [weekNumbers])
+
+  return (
+    <div
+      className={`
+      juno-datetimepicker-wrapper
+      ${width == "auto" ? "jn-inline-block" : "jn-block"}
+      ${width == "auto" ? "jn-w-auto" : "jn-w-full"}
+      ${wrapperClassName}
+    `}
+    >
+      <div className={`juno-datetimepicker-input-wrapper ${inputWrapperStyles}`}>
+        <input
+          className={`
+            juno-datetimepicker-input 
+            ${inputStyles}
+            ${label ? inputWithLabelStyles : inputWithoutLabelStyles}
+            ${isInvalid ? "juno-datetimepicker-input-invalid " + inputInvalidStyles : ""} 
+            ${isValid ? "juno-datetimepicker-input-valid" + inputValidStyles : ""}  
+            ${isValid || isInvalid ? "" : inputDefaultBorderStyles} 
+            ${width == "auto" ? "jn-w-auto" : "jn-w-full"}
+            ${enableTime && noCalendar ? "juno-datetimepicker-input-timepicker" : "juno-datetimepicker-input-default"}
+            ${className}
+          `}
+          data-mode={mode}
+          disabled={disabled}
+          id={theId}
+          name={name && name.length ? name : undefined}
+          onBlur={handleBlur}
+          onFocus={handleInputFocus}
+          placeholder={placeholder}
+          ref={fpRef}
+          type="text"
+          {...props}
+        />
+
+        {label && label.length ? (
+          <Label
+            text={label}
+            htmlFor={theId}
+            className={`${labelStyles}`}
+            disabled={disabled}
+            required={required}
+            floating
+            minimized={
+              placeholder || isOpen || theDate.selectedDate?.length || theDate.selectedDateStr?.length ? true : false
+            }
+          />
+        ) : (
+          ""
+        )}
+
+        <div className={`juno-datetimepicker-icon-container ${iconContainerStyles}`}>
+          {theDate.selectedDate?.length || theDate.selectedDateStr?.length ? (
+            <Icon icon="close" onClick={handleClearIconClick} disabled={disabled} title="Clear" />
+          ) : (
+            ""
+          )}
+          {isInvalid ? <Icon icon="dangerous" color="jn-text-theme-error" /> : ""}
+          {isValid ? <Icon icon="checkCircle" color="jn-text-theme-success" /> : ""}
+        </div>
+      </div>
+      <div ref={calendarTargetRef}></div>
+
+      {errortext && hasLength(errortext) ? <FormHint text={errortext} variant="error" className="jn-mt-0" /> : ""}
+      {successtext && hasLength(successtext) ? (
+        <FormHint text={successtext} variant="success" className="jn-mt-0" />
+      ) : (
+        ""
+      )}
+      {helptext && hasLength(helptext) ? <FormHint text={helptext} className="jn-mt-0" /> : ""}
+    </div>
+  )
+}
+
+// eslint-disable-next-line no-unused-vars
+type DateChangeHandler = (dates?: Date[], dateStr?: string, instance?: flatpickr.Instance) => void
+
+// eslint-disable-next-line no-unused-vars
+type LegacyEventHandler = (...args: any) => any
```

Comment:
use unknown isntead of any

Reply:
It won't allow for backward compatibility. Previously this was arbitrary function, I think 'any' closely matches such use case while 'unknown' doesn't. eg.

const OnYearChange = (_a :string) :void => { }

this gets accepted with 'any' (pre-migration case) and does not with 'unknown'